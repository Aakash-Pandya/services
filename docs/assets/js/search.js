var typedoc = typedoc || {};
            typedoc.search = typedoc.search || {};
            typedoc.search.data = {"kinds":{"1":"External module","32":"Variable","64":"Function","2097152":"Object literal"},"rows":[{"id":0,"kind":1,"name":"\"constants\"","url":"modules/_constants_.html","classes":"tsd-kind-external-module"},{"id":1,"kind":32,"name":"ASK_FOR_TYPE","url":"modules/_constants_.html#ask_for_type","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":2,"kind":32,"name":"ASK_FOR_LABEL","url":"modules/_constants_.html#ask_for_label","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":3,"kind":32,"name":"ASK_FOR_MESSAGE","url":"modules/_constants_.html#ask_for_message","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":4,"kind":32,"name":"labels","url":"modules/_constants_.html#labels","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":5,"kind":2097152,"name":"FEATURE","url":"modules/_constants_.html#feature","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":6,"kind":32,"name":"explanation","url":"modules/_constants_.html#feature.explanation-1","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FEATURE"},{"id":7,"kind":32,"name":"key","url":"modules/_constants_.html#feature.key-1","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FEATURE"},{"id":8,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#feature.needslabel-1","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FEATURE"},{"id":9,"kind":32,"name":"value","url":"modules/_constants_.html#feature.value-1","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FEATURE"},{"id":10,"kind":2097152,"name":"TEST","url":"modules/_constants_.html#test","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":11,"kind":32,"name":"explanation","url":"modules/_constants_.html#test.explanation-5","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TEST"},{"id":12,"kind":32,"name":"key","url":"modules/_constants_.html#test.key-5","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TEST"},{"id":13,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#test.needslabel-5","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TEST"},{"id":14,"kind":32,"name":"value","url":"modules/_constants_.html#test.value-5","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TEST"},{"id":15,"kind":2097152,"name":"FIX","url":"modules/_constants_.html#fix","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":16,"kind":32,"name":"explanation","url":"modules/_constants_.html#fix.explanation-2","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FIX"},{"id":17,"kind":32,"name":"key","url":"modules/_constants_.html#fix.key-2","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FIX"},{"id":18,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#fix.needslabel-2","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FIX"},{"id":19,"kind":32,"name":"value","url":"modules/_constants_.html#fix.value-2","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".FIX"},{"id":20,"kind":2097152,"name":"TYPINGS","url":"modules/_constants_.html#typings","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":21,"kind":32,"name":"explanation","url":"modules/_constants_.html#typings.explanation-6","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TYPINGS"},{"id":22,"kind":32,"name":"key","url":"modules/_constants_.html#typings.key-6","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TYPINGS"},{"id":23,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#typings.needslabel-6","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TYPINGS"},{"id":24,"kind":32,"name":"value","url":"modules/_constants_.html#typings.value-6","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".TYPINGS"},{"id":25,"kind":2097152,"name":"SUPPORT","url":"modules/_constants_.html#support","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":26,"kind":32,"name":"explanation","url":"modules/_constants_.html#support.explanation-4","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".SUPPORT"},{"id":27,"kind":32,"name":"key","url":"modules/_constants_.html#support.key-4","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".SUPPORT"},{"id":28,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#support.needslabel-4","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".SUPPORT"},{"id":29,"kind":32,"name":"value","url":"modules/_constants_.html#support.value-4","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".SUPPORT"},{"id":30,"kind":2097152,"name":"REFACTOR","url":"modules/_constants_.html#refactor","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":31,"kind":32,"name":"explanation","url":"modules/_constants_.html#refactor.explanation-3","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".REFACTOR"},{"id":32,"kind":32,"name":"key","url":"modules/_constants_.html#refactor.key-3","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".REFACTOR"},{"id":33,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#refactor.needslabel-3","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".REFACTOR"},{"id":34,"kind":32,"name":"value","url":"modules/_constants_.html#refactor.value-3","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".REFACTOR"},{"id":35,"kind":2097152,"name":"DOCS","url":"modules/_constants_.html#docs","classes":"tsd-kind-object-literal tsd-parent-kind-external-module","parent":"\"constants\""},{"id":36,"kind":32,"name":"explanation","url":"modules/_constants_.html#docs.explanation","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".DOCS"},{"id":37,"kind":32,"name":"key","url":"modules/_constants_.html#docs.key","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".DOCS"},{"id":38,"kind":32,"name":"needsLabel","url":"modules/_constants_.html#docs.needslabel","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".DOCS"},{"id":39,"kind":32,"name":"value","url":"modules/_constants_.html#docs.value","classes":"tsd-kind-variable tsd-parent-kind-object-literal","parent":"\"constants\".DOCS"},{"id":40,"kind":32,"name":"typesOfCommit","url":"modules/_constants_.html#typesofcommit","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":41,"kind":32,"name":"typesOfCommitKeys","url":"modules/_constants_.html#typesofcommitkeys","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":42,"kind":32,"name":"explanationOfTypes","url":"modules/_constants_.html#explanationoftypes","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"constants\""},{"id":43,"kind":1,"name":"\"modules/promptSelect\"","url":"modules/_modules_promptselect_.html","classes":"tsd-kind-external-module"},{"id":44,"kind":64,"name":"promptSelect","url":"modules/_modules_promptselect_.html#promptselect","classes":"tsd-kind-function tsd-parent-kind-external-module","parent":"\"modules/promptSelect\""},{"id":45,"kind":1,"name":"\"modules/getCommitLabel\"","url":"modules/_modules_getcommitlabel_.html","classes":"tsd-kind-external-module"},{"id":46,"kind":64,"name":"getCommitLabel","url":"modules/_modules_getcommitlabel_.html#getcommitlabel","classes":"tsd-kind-function tsd-parent-kind-external-module","parent":"\"modules/getCommitLabel\""},{"id":47,"kind":1,"name":"\"modules/getCommitType\"","url":"modules/_modules_getcommittype_.html","classes":"tsd-kind-external-module"},{"id":48,"kind":64,"name":"getCommitType","url":"modules/_modules_getcommittype_.html#getcommittype","classes":"tsd-kind-function tsd-parent-kind-external-module","parent":"\"modules/getCommitType\""},{"id":49,"kind":1,"name":"\"modules/promptInput\"","url":"modules/_modules_promptinput_.html","classes":"tsd-kind-external-module"},{"id":50,"kind":64,"name":"promptInput","url":"modules/_modules_promptinput_.html#promptinput","classes":"tsd-kind-function tsd-parent-kind-external-module","parent":"\"modules/promptInput\""},{"id":51,"kind":1,"name":"\"modules/showExplanations\"","url":"modules/_modules_showexplanations_.html","classes":"tsd-kind-external-module"},{"id":52,"kind":64,"name":"showExplanations","url":"modules/_modules_showexplanations_.html#showexplanations","classes":"tsd-kind-function tsd-parent-kind-external-module","parent":"\"modules/showExplanations\""},{"id":53,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-external-module"},{"id":54,"kind":64,"name":"commitMessage","url":"modules/_index_.html#commitmessage","classes":"tsd-kind-function tsd-parent-kind-external-module","parent":"\"index\""},{"id":55,"kind":1,"name":"\"debug\"","url":"modules/_debug_.html","classes":"tsd-kind-external-module"}]};