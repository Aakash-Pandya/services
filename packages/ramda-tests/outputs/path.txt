

  path
    ✓ takes a path and an object and returns the value at the path or undefined
    ✓ takes a path that contains indices into arrays
    1) takes a path that contains negative indices into arrays
    ✓ gets a deep property's value from objects
    2) returns undefined for items not found
    ✓ works with falsy items


  4 passing (66ms)
  2 failing

  1) path
       takes a path that contains negative indices into arrays:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ 'undefined'
- '"c"'
      + expected - actual

      -undefined
      +"c"
      
      at strictEqual (test/shared/eq.js:8:10)
      at Context.eq (test/path.js:42:5)
      at processImmediate (internal/timers.js:439:21)

  2) path
       returns undefined for items not found:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ 'null'
- 'undefined'
      + expected - actual

      -null
      +undefined
      
      at strictEqual (test/shared/eq.js:8:10)
      at Context.eq (test/path.js:54:5)
      at processImmediate (internal/timers.js:439:21)



