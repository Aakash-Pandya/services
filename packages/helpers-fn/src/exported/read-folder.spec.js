import { resolve } from 'path'
import { readFolder } from './read-folder.js'
const testDir = resolve(__dirname, '../../src')

test('happy', async () => {
  expect(await readFolder({ folder : testDir })).toMatchSnapshot()
})

test('with filter', async () => {
  const filterFn = x => x.endsWith('.spec.js')

  expect(await readFolder({
    folder : testDir,
    filterFn,
  })).toMatchSnapshot()
})

test('with exclude', async () => {
  const excludeFn = dir => dir.endsWith('log')

  expect(await readFolder({
    folder : testDir,
    excludeFn,
  })).toMatchSnapshot()
})
