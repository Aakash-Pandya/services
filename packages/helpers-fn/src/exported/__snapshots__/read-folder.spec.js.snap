// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Array [
  "/home/s/repos/services/packages/helpers-fn/src/index.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/createBenchmark.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/exec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/read-folder.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/read-folder.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/runTests.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/runTests.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/translate.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/translate.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/back.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/big.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/box.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/general.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/icon.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/iconSet.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/index.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/object.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/pattern.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/patternx.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/separator.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/spin.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/boxMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/figuresCollection.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/helpers.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/iconMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/iconSetMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/objMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/patternMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/patternxMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/separatorMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/spinMethod.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/modules/stopSpinMethod.js",
]
`;

exports[`with exclude 1`] = `
Array [
  "/home/s/repos/services/packages/helpers-fn/src/index.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/createBenchmark.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/exec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/read-folder.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/read-folder.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/runTests.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/runTests.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/translate.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/translate.spec.js",
]
`;

exports[`with filter 1`] = `
Array [
  "/home/s/repos/services/packages/helpers-fn/src/exported/read-folder.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/runTests.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/exported/translate.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/back.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/big.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/box.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/general.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/icon.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/iconSet.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/object.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/pattern.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/patternx.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/separator.spec.js",
  "/home/s/repos/services/packages/helpers-fn/src/log/spin.spec.js",
]
`;
